src/docker/dockerclient.cc

```diff
@@ -13,7 +14,7 @@ DockerClient::HttpClientFactory DockerClient::DefaultHttpClientFactory = []() {
 DockerClient::DockerClient(std::shared_ptr<HttpInterface> http_client) : http_client_{std::move(http_client)} {}
 
 bool DockerClient::serviceRunning(const std::string& app, const std::string& service, const std::string& hash) {
-  updateContainerStatus();
+  refresh();
```

Comment:
i hit submit on a comment right after your force-push update. i wonder if making this class stateful via `updateContainerStatus` is going to cause confusion like this in the future? or maybe its just a couple of places and its easy to reason about?

Reply:
AFAIU I think it makes sense to always work on the latest info (just as we would do if we were using ps); but we need to add a lock on the refresh method